index.js

```diff
@@ -201,6 +202,56 @@ module.exports = function(options) {
           return callback(null);
         });
       },
+      function gitTag(callback) {
+        if (!options.version) {
+          return callback(null);
+        }
+        const packageJson = files.reduce(function(path, file) {
+          if (file.path.indexOf('package.json') !== -1) {
+            path = file.path
+          }
+          return path
+        }, '');
+        if (packageJson === '') {
+          return callback('package.json is required if option version is true');
+        }
+        var packageContent = fs.readFileSync(packageJson, 'utf8');
+        var json = JSON.parse(packageContent.toString());
+        var version = json.version;
+        gutil.log(gutil.colors.yellow('Tag version ' + version));
+        var cmdTag = spawn('git', ['tag', '-a', 'v' + version, '-m', 'Version ' + version], {cwd: repoPath});
+        cmdTag.stderr.on('data', function(data) {
+          if (options.verbose || options.debug) gutil.log(gutil.colors.magenta('git tag: ') + data.toString().trim());
+        });
+        cmdTag.stdout.on('data', function(data) {
+          if (options.verbose || options.debug) gutil.log(gutil.colors.magenta('git tag: ') + data.toString().trim());
+        });
+        cmdTag.on('close', function(code) {
+          if (code !== 0) {
+            return callback('git tag exited with code ' + code);
+          }
+          return callback(null);
+        });
+      },
+      function gitPushTag(callback) {
```

Comment:
Attaching this comment to the new changeset since it turned outdated on the rename:

Can't the versioned tag be pushed out with the rest of the repository? Shouldn't be a need for multiple git push calls. Do correct me if I'm missing something though.

Since this is also located before the gitPush function, I'm pretty sure this will break usage of the remoteBranch option and push to the default branch regardless.